sdk/spring-sdk/src/main/java/kalix/springsdk/annotations/GenerateEntityKey.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kalix.springsdk.annotations;
+
+import java.lang.annotation.*;
+
+/**
+ * Indicates that Kalix should generate an entity key when this method is invoked.
+
+ * The generated entity key will be accessible via the entity context supplied in the SDK.
+ * A method annotated with this annotations should not be annotated with {@link EntityKey},
+ * if it does, an error will be raised.
+ *
+ * The generated key will be a Version 4 (random) UUID. The UUID will be generated using a cryptographically secure
+ * random number generator.
+ */
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+public @interface GenerateEntityKey {
+
+  Type value() default Type.VERSION_4_UUID;
+
+  enum Type {
+    /**
+     * Generate a Version 4 (random) UUID. The UUID will be generated using a cryptographically secure random
+     * number generator.
+     */
+    VERSION_4_UUID
```

Comment:
In gRPC, we need to have a value. In Java, we could just have an annotation. 

Not sure if we should keep this. Looking for ideas / feedback.

Reply:
I'm not sure I understand the problem.
`In gRPC, we need to have a value. In Java, we could just have an annotation.` I can't see any other elegant way to provide the users the ability to express that they want an auto-generated key. Another option would be adding a boolean property in the `Entity`? Such as `generateKey `?  But this doesn't match the current model of annotations in the SDK. Maybe if we were creating things with a [builder pattern](https://www.digitalocean.com/community/tutorials/builder-design-pattern-in-java) like in the java-SDK that would make sense. 
So all of that is to say that I would keep this implementation.

On top of that from what I understand this class `value` is never used and that is why you set it to a fixed value and with an enum with only one option. To inform the user this is it. There are no other options here. This makes sense to me. I quite like it.

Reply:
A gRPC annotation can't stands on its own. It needs to have an assigned value. Even though we only have one type, UUID V4, we need to write `option (kalix.method).entity.key_generator = VERSION_4_UUID`. 

The Java annotation could have been just `@GenerateEntityKey` without any value.
Basically, the enum I introduced is not needed and is there more to align with the gRPC counterpart. That's all.